__test__/lib/signerVerifier.test.js

```diff
@@ -12,46 +12,55 @@ const textEncoder = new TextEncoder();
 
 const DUMMY_MERKLE_ROOT = 'aa4149dda8fd2fac435898372f1de399140f6c50dbc3d40585c913701ce902c4';
 
-// TODO: Replace verify with new implementation once ready (IDCOM-1428)
-const verify = (data, signature, publicKey) => nacl.sign.detached.verify(
-  textEncoder.encode(data),
-  Uint8Array.from(Buffer.from(signature, 'hex')),
-  publicKey,
-);
-
 describe('signerVerifier', () => {
   beforeAll(mockDids);
 
   it('creates a signer from a private key', async () => {
-    const keypair = keyPair(DID_SPARSE);
+    const privateKey = privateKeyBase58(DID_SPARSE);
+    const verificationMethod = `${DID_SPARSE}#default`;
 
     const signer = await signerVerifier.signer({
-      verificationMethod: `${DID_SPARSE}#default`,
-      privateKey: privateKeyBase58(DID_SPARSE),
+      verificationMethod,
+      privateKey,
     });
 
     const signed = signer.sign({ merkleRoot: DUMMY_MERKLE_ROOT });
 
     expect(signed).toBeTruthy();
 
-    const verified = verify(DUMMY_MERKLE_ROOT, signed.signature, keypair.publicKey);
+    const verifier = await signerVerifier.verifier(DID_SPARSE, verificationMethod);
```

Comment:
Since DID_SPARSE is contained in verificationMethod, is this interface not redundant? Or could these values be different from each other? (e.g. I sign with MY verification method for YOUR DID)

Reply:
There is the case for the controller relationship... i.e my DID (from the VM) is a controller on your DID.

I added a test for this case (I only had one for signing)